src/massaction_rates.jl

```diff
@@ -4,7 +4,7 @@
 ###############################################################################
 
 @inbounds @fastmath function evalrxrate(speciesvec::AbstractVector{T}, rateconst,
-                                        stochmat::AbstractVector{Pair{T,T}})::typeof(rateconst) where T
+                                        stochmat::AbstractVector{Pair{Int,S}})::typeof(rateconst) where {T,S}
```

Comment:
I can make it Pair{S,V} to allow a third type if you think that might be useful. The first pair entry has to be a valid type for indexing (it indexes into the species population vector). The second is the stoichiometry type.

Reply:
`Int` is fine. I am just curious if `UInt8` indexing or something like that could be used to speed this up a bunch, and since it's simple to leave it open we might as well.

Reply:
I'll update it now to be parametric.